--- a/src/SFML/System/Android/Activity.cpp
+++ b/src/SFML/System/Android/Activity.cpp
@@ -29,9 +29,12 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/System/Android/Activity.hpp>
 #include <android/log.h>
+#include <cassert>

 #define LOGE(...) ((void)__android_log_print(ANDROID_LOG_INFO, "sfml-error", __VA_ARGS__))

+static const constexpr char* STATE_ERROR = "Bad state: there is probably no native activity running";
+
 LogcatStream::LogcatStream() :
 std::streambuf()
 {
@@ -63,6 +66,10 @@ ActivityStates* getActivity(ActivityStat
     if (!states || reset)
         states = initializedStates;

+    if (states == NULL && !reset) {
+        LOGE(STATE_ERROR);
+        throw std::runtime_error(STATE_ERROR);
+    }
     return states;
 }
 }
