--- a/src/SFML/Graphics/Image.cpp
+++ b/src/SFML/Graphics/Image.cpp
@@ -120,16 +120,7 @@ void Image::create(unsigned int width, u
 ////////////////////////////////////////////////////////////
 bool Image::loadFromFile(const std::string& filename)
 {
-    #ifndef SFML_SYSTEM_ANDROID
-
-        return priv::ImageLoader::getInstance().loadImageFromFile(filename, m_pixels, m_size);
-
-    #else
-
-        priv::ResourceStream stream(filename);
-        return loadFromStream(stream);
-
-    #endif
+    return priv::ImageLoader::getInstance().loadImageFromFile(filename, m_pixels, m_size);
 }
 
 
